### ISSUE TITLE
Memory Profiler Build

### ISSUE THREAD

**Issue by DennisSmolek** [pmndrs/react-three-jolt#54]
With memory being the #1 killer for Jolt projects we should really ensure that at least the library devs are able and know how to setup debug environments and running the memory profiler at the same time.

Jrouwe's recent comment on a bug I had points to the ability to run the profiler while the app is running.

[See the comment and what that looks like](https://github.com/jrouwe/JoltPhysics.js/discussions/149#discussioncomment-9095709)

- [x] Make sure the Memory Profiler build in the apps/examples folder is setup correctly
- [ ] Add the CSS to display the profiler while the app is running
- [ ] Test running the profiler with the app at the same time

**Comment by DennisSmolek** [comment]
_Automated code audit, 2026-09-20 — findings verified against the source at the cited lines._

Update from a fresh code audit (2026-09-20): confirming the vendored-build cleanup already implied by this issue's scope. `apps/examples/src/jolt/` is 1.0GB across 65 tracked files in 7 variant directories; only `Distribution/jolt-physics.wasm-compat` is actually imported (`BallBox.tsx:6`, `CharacterVirtualDemo.tsx:6`) -- the other ~985MB are referenced by nothing. Splitting the deletion off into its own issue, #163, since it's a quick dependency-free win that shouldn't wait on this issue's `?jolt=debug` profiler-switch feature. Keeping this issue scoped to the profiler switch itself (using jolt 1.1's official `debug-wasm-compat` subpath via `<Physics module>` instead of a hand-vendored build).
